src/main/java/app/gpx_animator/core/renderer/Renderer.java

```diff
@@ -527,14 +527,14 @@ private void trimGpxData(final TreeMap<Long, Point2D> timePointMap, final TrackC
     }
 
     private void keepFrame(@NonNull final List<RendererPlugin> plugins, @NonNull final RenderingContext rc,
-                               @NonNull final FrameWriter frameWriter, @Nullable final BufferedImage bi, final int frames,
-                               @NonNull final TreeMap<Long, Point2D> wpMap, @Nullable final Long keepFrame) throws UserException {
+                           @NonNull final FrameWriter frameWriter, @Nullable final BufferedImage bi, final int frame,
+                           @NonNull final TreeMap<Long, Point2D> wpMap, @Nullable final Long keepFrame) throws UserException {
         if (bi != null && keepFrame != null && keepFrame > 0) {
-            drawWaypoints(bi, frames, wpMap);
-            final var marker = drawMarker(bi, frames);
+            drawWaypoints(bi, frame, wpMap);
+            final var marker = drawMarker(bi, frame);
 
             for (final var plugin : plugins) {
-                plugin.renderFrame(frames, marker, bi);
+                plugin.renderFrame(frame, marker, bi);
```

Comment:
Why did you rename this variable? The `frame` variable contains the number of the actual frame to render, not multiple frames.